### ISSUE TITLE
Assert failure while loading customizer preset

### ISSUE THREAD

**Issue by nyponen** [openscad/openscad#2461]
This occurred while loading a customizer preset on OpenSCAD version 2018.04.01 (git e7c0851)

![loadpresetassert](https://user-images.githubusercontent.com/17462467/44629399-9edbc880-a903-11e8-99a0-642940647c93.jpg)

**Comment by t-paul** [comment]
Please add some example files to reproduce.

**Comment by nyponen** [comment]
This occurs randomly while switching between different presets. I can't reproduce it at will.

**Comment by t-paul** [comment]
So it happens with different models at random times when changing the preset selection box to a different value?

**Comment by nyponen** [comment]
> So it happens with different models at random times when changing the preset selection box to a different value?

Yes.

**Comment by t-paul** [comment]
Please check with latest snapshot version which has a fix that could be related (#2404).

**Comment by nyponen** [comment]
I have a different error with the latest snapshot. When I open a preset, an asterisk appears next to the name in the drop down list, and if I try to load a different preset, I get this dialog box, even though I made no changes to the preset. When I tried to simplify the code to make a test case, the problem went away.

![preseterror](https://user-images.githubusercontent.com/17462467/44631015-05201580-a91b-11e8-9a1e-551487343a59.JPG)

**Comment by t-paul** [comment]
Maybe the code has a new variable that is not yet in the presets? Would it be possible to share that original code?

**Comment by nyponen** [comment]
No, the code didn't have any new variables in it. This simplified code shows the problem on my WIndows 10 machine. Here's the .scad file:
`/* [1] */
Element1 = true;

/* [2] */
Elements = 12;              // [1:1:34]   

design( 
    element1 = Element1 
);`

Here's the .json file:
`{
    "parameterSets": {
        "Feb22": {
            "Elements": "13",
            "outer_diameter": "30"
        }
    },
    "fileFormatVersion": "1"
}
`
